sim/hunter/serpent_sting.go

```diff
@@ -99,7 +101,10 @@ func (hunter *Hunter) chimeraShotSerpentStingSpell(rank int) *core.Spell {
 		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
 			// As of phase 5 the only time serpent sting scales with AP is using the Dragonstalker's Pursuit 6P - this AP scaling doesn't benefit from target AP modifiers
 			damage := baseDamage + (hunter.SerpentStingAPCoeff * spell.RangedAttackPower(target, true))
-			spell.CalcAndDealDamage(sim, target, damage, spell.OutcomeRangedHitAndCrit)
+
+			spell.WaitTravelTime(sim, func(s *core.Simulation) {
+				spell.CalcAndDealDamage(sim, target, damage, spell.OutcomeRangedHitAndCrit)
```

Comment:
Actually quick thought here. The game calculates damage at cast time, so you should calculate the damage before waiting then deal the damage after waiting.

```
result := spell.CalcDamage(sim, target, damage, spell.OutcomeRangedHitAndCrit)
spell.WaitTravelTime(sim, func(s *core.Simulation) {
  result.DealDamage()
})
```